src/Microsoft.AspNetCore.Mvc.ViewFeatures/Controller.cs

```diff
@@ -102,7 +102,7 @@ public dynamic ViewBag
         [NonAction]
         public virtual ViewResult View()
         {
-            return View(viewName: null);
+            return View(viewName: ControllerContext.ActionDescriptor.ActionName);
```

Comment:
Curious, the bug doesn't say about these changes..why are you making them? also one could return view result from a filter too in which case the earlier behavior of defaulting to the currently action name seems fine.